base/math.jl

```diff
@@ -1245,7 +1245,18 @@ end
 function ^(x::Union{Float16,Float32}, n::Integer)
     n == -2 && return (i=inv(x); i*i)
```

Comment:
cases `n == 0`, `n == 1`and `n == -1` perform unnecessary conversions (widen and recast).